Dayantur/precheck dev

dayantur opened the pull request:
**Summary**
This PR:

- adds a new precheck function: precheck_warn_zero_sfr_params, designed to print a Supy INFO for each site and each land cover surface with sfr == 0. The function lists all land cover parameters that have a non-null value and were not physically prechecked, flagging them as potentially unused or inconsistent. Note: no test has been added for this function since it only prints supy infos!
- adds some YAML test cases and a precheck_testcase.py script in test/precheck_testcase to test precheck procedure when stebbsmethod == 0.
- adds docstrings to all defs for run_precheck() in core.py

sunt05 commented:
# Draft Comment for PR #446

Hi @dayantur,

Thanks for your continued work on the conditional validation! I've just posted some suggestions in issue #400 about adapting to the new top-down validation approach we've implemented to fix the false positives (see #444 and #448).

Would you mind taking a look at the comment I posted there? It has detailed migration suggestions that might be helpful for updating this PR to work with the new validation structure.

If the approach makes sense to you, we could coordinate on how to best integrate these changes. Happy to help with any questions or discuss alternative approaches if needed!

dayantur commented:
Following the discussion in issue #400  -  this is a complete summary of this PR:

- precheck procedure and Pydantic validation are now decoupled: I've moved precheck logic from `core.py` into `precheck.py` and updated all tests in `test/test_precheck.py`. 
- YAML test cases for precheck testing are collected in `test/precheck_testcase`
- functions in `run_precheck()` have now docstrings
- a new function (`precheck_warn_zero_sfr_params `) has been added to `run_precheck()` to show names of parameters that are not preprocessed during the precheck (since associated with a sfr == 0)

sunt05 reviewed (commented):
Hi @dayantur, please review the ignored files to ensure no necessary tests are excluded. Otherwise, this looks good to me. Once confirmed, we can merge and move forward. Well done!
